### PR TITLE
chore: update node to 18 lts

### PR DESCRIPTION
**Pull request by stevenh**
Update to the node 18 LTS so we're on a supported version.

Also updating a bunch of logging to help investigate the random failures occurring in workflows, including switching CodeNarcServer to use java logging instead of prints.

### CODE REVIEW

**Comment by nvuillam** [comment]
@stevenh can it be merged ? :)

Please remove the "draft" status of the PRs you consider ready ;) (i think this one is ^^ )

File: groovy/src/main/com/nvuillam/CodeNarcServer.groovy [hunk 1]
```diff
@@ -32,13 +32,21 @@ import org.apache.tools.ant.types.Commandline
 // CodeNarc main class
 import org.codenarc.CodeNarc
 
+// Logging
```
Comment (nvuillam) [inline groovy/src/main/com/nvuillam/CodeNarcServer.groovy:35]: such an elegant rewriting ;)

**Comment by stevenh** [comment]
> @stevenh can it be merged ? :)
> 
> Please remove the "draft" status of the PRs you consider ready ;) (i think this one is ^^ )

Will do, was just fixing the spelling issues.

I do wish we wouldn't keep getting random failures on some of the platforms.

**Comment by nvuillam** [comment]
are the failures so random ? ^^

**Comment by stevenh** [comment]
> are the failures so random ? ^^
Yes it seems so, I'd say they currently fail more than succeed, from what I've seen while pushing up incremental fixes.

**Comment by stevenh** [comment]
There's a few ready for review now, will rebase the others when this is in as that will help identify any flaky behaviour.

**Review by nvuillam — APPROVED** [review]
Great 🔥🔥🔥🔥

**Pull request merged**